packages/fetchai/skills/generic_buyer/handlers.py

```diff
@@ -215,9 +216,11 @@ def _handle_match_accept(
             ledger_api_dialogue = cast(LedgerApiDialogue, ledger_api_dialogue)
             ledger_api_dialogue.associated_fipa_dialogue = fipa_dialogue
             fipa_dialogue.associated_ledger_api_dialogue = ledger_api_dialogue
-            self.context.outbox.put_message(message=ledger_api_msg)
-            self.context.logger.info(
-                "requesting transfer transaction from ledger api..."
+            tx_behaviour = cast(
+                GenericTransactionBehaviour, self.context.behaviours.transaction
+            )
+            tx_behaviour.waiting.append(
```

Comment:
A comment: `append` is fine because handlers and behaviours are processed sequentially or asynchronously by the scheduler.
It would also works in multithreading, because in CPython primitive operations over data structures are thread-safe.

Reply:
Also, even if using multithreading; skills are always executed in the same thread atm.

Reply:
true, I had in mind an extension where many threads could write to it, e.g. one day we could have more threads for a skill

Reply:
ok but then we need to revisit other parts of the code too :)